github/resource_github_repository_ruleset.go

```diff
@@ -0,0 +1,404 @@
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"dismiss_stale_reviews_on_push": {
+							Type:        schema.TypeBool,
+							Required:    true,
+							Description: "New, reviewable commits pushed will dismiss previous pull request review approvals.",
+						},
+						"require_code_owner_review": {
+							Type:        schema.TypeBool,
+							Required:    true,
+							Description: "Require an approving review in pull requests that modify files that have a designated code owner.",
+						},
+						"require_last_push_approval": {
+							Type:        schema.TypeBool,
+							Required:    true,
+							Description: "Whether the most recent reviewable push must be approved by someone other than the person who pushed it.",
+						},
+						"required_approving_review_count": {
+							Type:         schema.TypeInt,
+							Required:     true,
+							Description:  "The number of approving reviews that are required before a pull request can be merged.",
+							ValidateFunc: validation.IntBetween(0, 10),
+						},
+						"required_review_thread_resolution": {
+							Type:        schema.TypeBool,
+							Required:    true,
+							Description: "All conversations on code must be resolved before a pull request can be merged.",
+						},
+					},
+				},
+			},
+			"rule_required_status_checks": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				ForceNew:    true, // TODO: Remove this when updating is implemented
+				MaxItems:    1,
+				Description: "Choose which status checks must pass before branches can be merged into a branch that matches this rule. When enabled, commits must first be pushed to another branch, then merged or pushed directly to a branch that matches this rule after status checks have passed.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"strict_required_status_checks_policy": {
+							Type:        schema.TypeBool,
+							Required:    true,
+							Description: "Whether pull requests targeting a matching branch must be tested with the latest code. This setting will not take effect unless at least one status check is enabled.",
+						},
+						"required_status_checks": {
+							// This field is based on "checks" in github_branch_protection_v3, which combines the context and integration_id (or app_id which it's called in branch_protection rules) into one string
+							Type:        schema.TypeSet,
+							Required:    true,
+							Description: "The list of status checks to require in order to merge into this branch. No status checks are required by default. Checks should be strings containing the 'context' and 'integration_id' like so 'context:integration_id'. Also supports only the 'context'",
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+						},
+					},
+				},
+			},
+			// TODO: Broken until a bump in the underlying sdk
```

Comment:
As mentioned in the PR description, this one is broken currently. `go-github` needs a release bump. As opposed to the `rule_update` field, I'm confident will be solved by this bump 😓